empirical-org/Empirical-Core
# Synthetic Labeled Data Uploader (#9641)

* Working version of array-based implementation

* Add instance based methods, more context in results.

* Remove old files.

* Rename some items.

* Adjust comment.

* Switch some method names, add basic specs.

* Fix typo, pass through options.

* Add another comment.

* Working version of training export.

* Extract out common methods.

* Clean up some methods.

* Changes so we only test/validate using real data.

* Add minimum for test, validation, train per label

* Confirm that labels have minimums.

* Rename LANGUAGES, since not used as default

* Reorganize file and rename for adding spelling.

* Add some more data size checks.

* Working version of spelling substitute.

* Simplify spelling code

* Updates to data script.

* Small adjustments based on testing.

* Make manual_types optional.

* Extract out manual types since it is not used much

* Rename some variables in ManualTypes.

* Move files to app directory

* Add some basic tests.

* Add spelling test.

* Allow for multiple spelling substitutions.

* Reasonably tested version of data generation.

* Small refactor.

* Use spellings instead of misspellings

* Move test_percent to manual types

* Passing tests.

* Move data_count to manual types.

* Don’t memoize data count.

* Lint.

* Add some basic tests for manual types.

* Small cleanup.

* Spelling changes.

* Linting changes.

* Make test more efficient.

* Update gem since it’s been a while.

* Fix test expectation for manually typing.

* Wip, breaking out the translation module.

* Add spelling class.

* Refactor and cleanup of data file.

* Comments and listing.

* Refactor result.

* Move expects into tests.

* lint

* Add TODO.

* Code golf.

* Whitespace.

* Pass in keys for generators.

* Use ApplicationService for Data.

* Move languages option to translation model.

* Update comment.

* Update comment.

* Use shorter method from PR feedback.

* Move translate gem to engine.

* Move files to engine.

* Namespace evidence, Add yml parser.

* Rename class

* More renaming.

* Delete test constants.

* More renaming.

* Remove unused methods.

* Frontend form for uploads.

* Add Route for staff csv uploads.

* Add a basic test for uploads.

* Add uploader references.

* Add basic data worker.

* Remove unused methods, setup main run method.

* Lower validations, and move into manual types.

* Use shared email address.

* Add basic endpoint.

* Test for background job.

* Add test for controller endpoint.

* Add csv creator test.

* Make email a variable to make testing easier.

* Pin dev to port 3000.

* Naming change.

* Reorder method.

* Move away from raises and use validations.

* Lint.

* Lint.

* Require google translate.

* Refactor for complexity.

* Handle some response errors.

* Small Style changes.

* Lint.

* Lint

## WHAT

## WHY

## HOW

### Screenshots
(If applicable. Also, please censor any sensitive data)

### Notion Card Links
(Please provide links to any relevant Notion card(s) relevant to this PR.)

PR Checklist | Your Answer
------------ | -------------
Have you added and/or updated tests? |  (The answer should mostly be 'YES'. If you answer 'NO', please justify.)
Have you deployed to Staging? | (Possible answers: YES, Not yet - deploying now!, NO - non-app change, NO - tiny change)
Self-Review: Have you done an initial self-review of the code below on Github? |
Spec Review: Have you reviewed the spec and ensured this meets requirements and/or matches design mockups? | (N/A or Yes)
Back-to-school 2022: Have you checked the [webinar schedule](https://www.notion.so/quill/Back-to-school-webinar-banners-2022-a75a89cfad9f434899ef6be3eb184733) to avoid for downtime/risky deploys? | (Yes or N/A)
